### DISCUSSION TITLE
Grid overview of student progress per level

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#4008] [Archive]
In the interest of making Hedy more usable in the classroom (see also #3930) I would love to see this feature that I have used "analog" in school for a while.

I place all names on a whiteboard, and all adventures. If a kid completes the adventure, I tick it off. This makes it easy to see progress, see this image (without names):

<img width="1012" alt="image" src="https://user-images.githubusercontent.com/1003685/216375864-d67f19e5-b015-425e-90f8-8a3abcf1cf0a.png">

It would be amazing if we can integrate this into Hedy! It is not entirely clear though what "completes" means if we want to automate it, but we can deal with that problem later. 

A first, fantastic, idea by @MarleenGilsing is to allow students to mark their own adventure as completed, see [Discussion](https://github.com/hedyorg/hedy/discussions/3930#discussioncomment-4852230_).

![image](https://user-images.githubusercontent.com/80678586/216350044-9dd17f0a-e885-40c1-be3c-c198370fa07c.png)

Once we have the tick marks, we can make it more extensive, f.e. a tick mark can link to the program so the teacher can see it or... in a later stage, we can add requirements for "auto completing" like the presents of a certain keyword and/or output.

**Comment by jpelay** [comment]
Hello miss @Felienne! I'd be very much interested in working on this! I have a few questions first: in this context you're using adventures and assignments interchangeably? Are all adventures assignments here? We can totally add the check marks with ease, and link them to programs for teachers to review. A further question about that: the linked program would be the last run or the one they had when they marked the assignment as completed?

**Reply by Felienne** [reply]
Hi @jpelay!

> Hello miss @Felienne! I'd be very much interested in working on this! I have a few questions first: in this context you're using adventures and assignments interchangeably? Are all adventures assignments here? 

Good catch, I made the text into adventures everywhere above because that is indeed what I meant.

> We can totally add the check marks with ease, and link them to programs for teachers to review. A further question about that: the linked program would be the last run or the one they had when they marked the assignment as completed?

Ah yes good question. I want to change the save interface anyway (why are we asking for save anyway, why don't we just autosave like any normal browser program?) but for now we could say that if the kid clicks the tick mark, the program is saved and that link is send to the teacher to review.

I was just teaching the other day, and I have another, even easier, idea for classroom teaching. The setting here is specifically that I walk around, and kids tell me when they are ready. I look at their program and if I deem it correct, I now walk to the whiteboard to write down a tickmark. But what if I can mark the program as correct FROM THEIR LAPTOPS? That would make it very easy for me!

It could work like this. 

* Teacher creates a pincode or password 
* Kid clicks the tick mark, gets a password prompt. I come by and fill it out
* Program is immediately ticked off

This would save me like 100 back and forths! Of course later we can extend/update it to the tickmark means the program is being handed in and shown in a grid to the teacher for them to view on their computer. That can be the default also when to password is set.

Hope this make sense!

Also, we have a student interested in working on this, but there is enough work I think for you both, and I would love to try out the password thing this academic year (the student could then do the submit and teacher reviews on their PC version)
